controllers/azuremanagedcontrolplane_controller.go

```diff
@@ -209,6 +189,26 @@ func (amcpr *AzureManagedControlPlaneReconciler) Reconcile(ctx context.Context,
 		}
 	}()
 
+	// Return early if the object or Cluster is paused.
+	if annotations.IsPaused(cluster, azureControlPlane) {
+		log.Info("AzureManagedControlPlane or linked Cluster is marked as paused. Won't reconcile normally")
+		return amcpr.reconcilePause(ctx, mcpScope)
+	}
+
+	// check if the control plane's namespace is allowed for this identity and update owner references for the identity.
+	if azureControlPlane.Spec.IdentityRef != nil {
+		err := EnsureClusterIdentity(ctx, amcpr.Client, azureControlPlane, azureControlPlane.Spec.IdentityRef, infrav1.ManagedClusterFinalizer)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+	} else {
+		warningMessage := ("You're using deprecated functionality: ")
```

Comment:
This could be one string literal, which is maybe more readable:

```go
warningMessage := "You're using deprecated functionality: " +
        "Using Azure credentials from the manager environment is deprecated and will be removed in future releases. " +
        "Please specify an AzureClusterIdentity for the AzureManagedControlPlane instead, see: https://capz.sigs.k8s.io/topics/multitenancy.html"
```